kratos/mpi/utilities/gather_modelpart_utility.cpp

```diff
@@ -0,0 +1,208 @@
+#include "mpi/utilities/gather_modelpart_utility.h"
+#include "mpi/utilities/parallel_fill_communicator.h"
+#include "includes/data_communicator.h"
+#include "mpi/includes/mpi_communicator.h"
+
+namespace Kratos
+{
+
+GatherModelPartUtility::GatherModelPartUtility(int gather_rank,
+                                               ModelPart& origin_model_part,
+                                               int mesh_id,
+                                               ModelPart& destination_model_part)
+    : mr_model_part(destination_model_part), mgather_rank(gather_rank)
+{
+    KRATOS_TRY;
+
+    const int mpi_rank = DataCommunicator::GetDefault().Rank();
```

Comment:
Please use the DataCommunicator of the origin_model_part:
`const DataCommunicator& r_comm = origin_model_part.GetCommunicator().GetDataCommunicator();`
`const int mpi_rank = r_comm.Rank();`

and so on where the DataCommunicator is used.